### DISCUSSION TITLE
CORS Options requests on Edge Functions sometimes randomly rejecting with error 500

### DISCUSSION THREAD

**Discussion by emilthemaker** [supabase/supabase#21949] [Questions]
My Edge Function is sometimes (seemingly randomly) returning 500 on CORS options requests without even booting up the function once. I can't see anything in logs, only the 500 response. Notably, this only appears to happen for some requests.

**Screenshots**
![CleanShot 2024-03-11 at 18 50 47@2x](https://github.com/supabase/supabase/assets/138380015/c82f506a-9135-4cca-94b7-8f0c7b955a3a)
Browser console of a user with the issue (can't replicate it)

![CleanShot 2024-03-11 at 18 51 43@2x](https://github.com/supabase/supabase/assets/138380015/b6329d01-8ddd-49fa-9497-76b2f62d9e6b)
Same request on the backend

(not sure if I can provide the request metadata here for security reasons)

**Comment by akabutnicer** [comment]
What might be occurring is rate limiting. In this case, I'd recommend you try and remake your supabase application.
